### ISSUE TITLE
`no-unsafe-string-replacement`: Allow statically safe `String#repeat()` replacements

### ISSUE THREAD

**Issue by Sli4o** [sindresorhus/eslint-plugin-unicorn#3693]
### Summary

The `no-unsafe-string-replacement` rule currently reports replacement strings produced by `String.prototype.repeat()`, even when the result is statically known to contain no replacement patterns.

### Code example

```js
text.replaceAll('\t', ' '.repeat(4));
```

The replacement expression always evaluates to four spaces. It cannot contain special replacement patterns such as `$&`, `$1`, or `$<name>`.

### Proposed solution

Consider allowing `String#repeat()` calls when the receiver is a string literal that is itself a safe replacement string and the repeat count can be statically evaluated.

For example, the following should be allowed:

```js
text.replaceAll('\t', ' '.repeat(4));
text.replaceAll('\t', '\u00A0'.repeat(4));
```

This would preserve the rule's protection against dynamic or potentially unsafe replacement strings while avoiding false positives for expressions whose result can be proven safe through static analysis.
